File: 8. String to Integer (atoi)/myAtoi.py
import re
class myAtoi:
    def solution(self, str: str) -> int:
        str = str.strip()
        if not str:
            return 0
        res = re.compile(r'^[\-\+]?\d+').findall(str)
        ans =  int(res[0]) if res else 0
        if ans > 0:
            ans = min(ans, 2147483647)
        elif ans < 0:
            ans = max(ans, -2147483648)
        return ans

obj = myAtoi()
string = "-91283472332"
res = obj.solution(string)
print(res)

File: 8. String to Integer (atoi)/myAtoi.java
class myAtoi {
    public int solution(String str) {
    	char[] chars = str.toCharArray();
    	int idx = 0;
    	while (idx < chars.length && chars[idx] == ' ') {
    		idx ++;
    	}
    	if (idx == chars.length) {
    		return 0;
    	}
    	boolean neg = false;
    	if (chars[idx] == '-') {
    		neg = true;
    		idx ++;
    	}
    	else if (chars[idx] == '+') {
    		idx ++;
    	}
    	else if ((chars[idx] - '0') < 0 && (chars[idx] - '0') > 9) {
    		return 0;
    	}
    	int num = 0;
    	while(idx < chars.length && (chars[idx] - '0') >= 0 && (chars[idx] - '0') <= 9) {
    		int n = chars[idx] - '0';
    		if (num > (Integer.MAX_VALUE - n) / 10) {
    			return neg ? Integer.MIN_VALUE : Integer.MAX_VALUE;
    		}
    		num = num * 10 + n;
    		idx ++;
    	}
    	return neg ? -num : num;

    }

    public static void main(String[] argv) {
    	String str = "   -42";
    	myAtoi obj = new myAtoi();
    	int res = obj.solution(str);
    	System.out.println(res);
    }
}